File: vapi/internal/PasswordService/PasswordService.go
package passwordservice

import (
	"golang.org/x/crypto/bcrypt"
)

type PasswordService interface {
	Hash(password string) (string, error)
	Verify(password, hashed string) bool
}

type BcryptPasswordService struct {
	cost int // default 12
}

func NewBcryptPasswordService(cost int) *BcryptPasswordService {
	if cost <= 0 {
		cost = 12
	}
	return &BcryptPasswordService{cost: cost}
}

func (s *BcryptPasswordService) Hash(password string) (string, error) {
	bytes, err := bcrypt.GenerateFromPassword([]byte(password), s.cost)
	return string(bytes), err
}

func (s *BcryptPasswordService) Verify(password, hashed string) bool {
	err := bcrypt.CompareHashAndPassword([]byte(hashed), []byte(password))
	return err == nil
}

File: vdotnet/VDotnetOrmTest/PasswordService.cs
using BCrypt.Net;
public interface IPasswordService
{
    string Hash(string password);
    bool Verify(string password, string hashed);
}


public class BcryptPasswordService : IPasswordService
{
    private readonly int _workFactor;

    public BcryptPasswordService(int workFactor = 12)
    {
        _workFactor = workFactor > 0 ? workFactor : 12;
    }

    public string Hash(string password)
    {
        return BCrypt.Net.BCrypt.HashPassword(password, _workFactor);
    }

    public bool Verify(string password, string hashed)
    {
        return BCrypt.Net.BCrypt.Verify(password, hashed);
    }
}
